### PR TITLE
boards: cc3220sf_launchxl: Make cc3220sf XIP by default

### PR DESCRIPTION
**Pull request by GAnthony**
Previously, the cc3220sf platform could only be flashed using
TI UniFlash (or CCS) tools, requiring unique instructions.
Also, these procedures were not integrated with the standard Zephyr
build targets: 'flash', 'debug', 'debugserver'.

After installation of a TI OpenOCD v1.10 release, and other
one-time board setup, flashing and debugging can now be done
via OpenOCD/gdb, using the standard Zephyr build targets per
the Zephyr Application Development primer.

This also makes the cc3220sf an XIP platform by default, and removes
some now extraneous board support files.

### CODE REVIEW

**Comment by GAnthony** [comment]
I believe OpenOCD is developed using Gerrit.   The TI patches are in review, and according to the developer, it will be several months before getting upstreamed, which is why the board.rst prerequisites instruct to git clone the TI OpenOCD repo (git.ti.com/sdo-emu/).

File: boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst [hunk 1]
```diff
@@ -150,65 +146,70 @@ Prerequisites:
...
-Studio IDE, and required for the dslite.sh tool.
+.. zephyr-app-commands::
+   :zephyr-app: <my_app>
+   :board: cc3220sf_launchxl
+   :goals: flash
 
-To see program output from UART0, one can execute in a separate terminal
+This will load the image into flash.
+
+To see program output from UART0, one can connect a separate terminal
```
Comment (dbkinder) [inline boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst]: delete ``one can``

File: boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst [hunk 2]
```diff
@@ -150,65 +146,70 @@ Prerequisites:
...
+   See Section 21.10 "Debugging Flash User Application Using JTAG" of the
+   `CC3220 TRM`_ for details on the secure flash boot process.
 
-Flashing Command:
------------------
 
-Once the above prerequisites are met, use the UniFlash command line tool
-to flash the Zephyr image:
+Once the above prerequisites are met, applications for the ``_cc3220sf_launchxl``
+board can be built, flashed and debugged with openocd and gdb per the Zephyr
```
Comment (dbkinder) [inline boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst]: ``built, flashed, and debugged``  (missing comma)

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looks good, a couple of small issues

File: boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst (hunk 2 above)
Comment (GAnthony) [reply boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst]: ah, the Oxford comma ;-)
OK, thanks for review.

**Comment by galak** [comment]
> I believe OpenOCD is developed using Gerrit. The TI patches are in review, and according to the developer, it will be several months before getting upstreamed, which is why the board.rst prerequisites instruct to git clone the TI OpenOCD repo (git.ti.com/sdo-emu/).

I guess I was asking to see if we could possibly pull the various changes into the Zephyr SDK so its openocd would support the board.

**Comment by GAnthony** [comment]
> I guess I was asking to see if we could possibly pull the various changes into the Zephyr SDK so its openocd would support the board

Given the patches are still under review, and the TI OpenOCD repo may be updated at a higher frequency than the Zephyr SDK, I would suggest it might be safer to take the TI OpenOCD install.  
My concern is the patches could break something for a different platform, a scenario which only gets tested (presumably) on official OpenOCD releases.

This separate install would be a temporary prerequisite, until the patches finally get merged.

File: boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst (hunk 1 above)
Comment (GAnthony) [reply boards/arm/cc3220sf_launchxl/doc/cc3220sf_launchxl.rst]: @dbkinder: Done.  Thanks, pushed updates.

**Review by dbkinder — APPROVED** [review]
thanks!

**Review by galak — APPROVED** [review]

**Pull request merged**